model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,202 @@
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'output:0', 'shape')
+            else:
+                PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:0', 'shape')
+                PermuteInputs().set_input_permutation(node.in_node(3), node, 'input:0', 'shape')
+
+    def normalize_strided_slice(self, graph: Graph, node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        input_rank = len(input_shape)
+        slice_rank = node.in_port(1).data.get_shape()[0]
+
+        slice_mask_names = ['begin_mask', 'end_mask', 'new_axis_mask', 'shrink_axis_mask', 'ellipsis_mask']
+        # allign masks sizes with slice_rank (not confuse with extending mask_alligment != mask_extending)
+        for mask_name in slice_mask_names:
+            num = slice_rank - len(node[mask_name])
+            val = 0 if mask_name not in ['begin_mask', 'end_mask'] else 1  # extend with ones only for begin and end
+            node[mask_name] = np.append(node[mask_name], [val] * num).astype(int)
+
+        if np.any(node.ellipsis_mask):
+            idx = np.nonzero(node.ellipsis_mask)
+            assert len(idx[0]) == 1, 'only one ellipsis_mask nonzero value is allowed'
+            ellipsis_start = idx[0][0]
+            # since we don't use begin, end values
+            node.begin_mask[ellipsis_start] = 0
+            node.end_mask[ellipsis_start] = 0
+
+            num = input_rank - slice_rank + np.count_nonzero(node.new_axis_mask[ellipsis_start:])
+
+            # unroll ellipsis for masks
+            node.ellipsis_mask[ellipsis_start] = 0
+            for mask_name in slice_mask_names:
+                node[mask_name] = np.insert(node[mask_name], ellipsis_start + 1, [0] * num).astype(int)
+
+            self.unroll_ellipsis_for_inputs(graph, node, ellipsis_start, num)
+        elif slice_rank - np.count_nonzero(node.new_axis_mask) < input_rank:
+            num = input_rank - slice_rank + np.count_nonzero(node.new_axis_mask)
+            self.extend_inputs(node, num)
+
+            # extend masks
+            for mask_name in slice_mask_names:
+                num = input_rank - len(node[mask_name]) + np.count_nonzero(node.new_axis_mask)
+                node[mask_name] = np.append(node[mask_name], [0] * num).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_ellipsis_ext: int):
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            if ellipsis_start != 0:
+                split = create_op_with_const_inputs(graph, VariadicSplit, {1: 0, 2: int64_array([ellipsis_start, -1])},
+                                                    {'name': node.name + '/split_to_unroll_{}_ellipsis'.format(slice_name),
```

Comment:
Introduce variable `node_name = node.soft_get('name', node.id)` and use it everywhere instead of `node.name`

Reply:
done